app/dojo.py

```diff
@@ -248,3 +251,73 @@ def print_unallocated(self,filename=None):
 			txt_file.write(output)
 			txt_file.close()
 			return ("\nData has been successfully saved to {}.txt\n" .format(filename))
+
+	def reallocate_person(self, emailaddress, roomname):
+		pass
+		"""This method reallocates a person using their unique identifier,
+		in this case their email address, to a the specified new room
+
+		unique_identifier = emailaddress
+		found_room = False
+		found_emailaddress = False
+
+		#checks if the emailaddress exists and returns the persons name associated
+		#with it
+		for room in itertools.chain(self.all_offices, self.all_livingspace):
+			if room.name == roomname:
+				if room.room_type == "OFFICE" and len(room.occupants) < 6:
+					new_room_allocation_type = room.room_type
+					new_room_allocation_name = room.name
+					found_room == True 
+				elif room.room_type == "LIVING SPACE" and len(room.occupants) < 4:
+					new_room_allocation_type = room.room_type
+					new_room_allocation_name = room.name 
+					found_room == True
+
+
+
+
+		if found_room:
+			for this_room in itertools.chain(self.all_offices, self.all_livingspace):
+				for occupant in this_room.occupants:
+					if occupant.email == emailaddress:
+						person_reallocating = occupant
+						print ("email address {} associated with {} {} has been found."
+							.format(emailaddress, occupant.role, occupant.name))
+						current_room_allocation_type = this_room.room_type
+						current_room_allocation_name =this_room.name"""
+
+	def load_people(self, filename):
+		"""This method adds people to rooms from a txt file.
+		The text input from the text file should have the following format
+		NAME EMAIL ROLE ACCOMODATION_OPTION
+
+		"""
+		if os.path.isfile(filename + ".txt"):
+			if os.stat(filename + ".txt").st_size:
+				with open(filename + ".txt") as input_file:
+					for line in input_file:
+						read_line = line.split()
+						if len(read_line) > 4 or len(read_line) < 3:
+							print (text_format.CRED + "\nInvalid entry!\n" + text_format.CEND)
+						try:
+							name = read_line[0]
+							email = read_line[1]
+							role = read_line[2]
+							accomodation_option = read_line[3]
+						except IndexError:
+							accomodation_option = "N"
+
+						self.add_person(name, email, role, accomodation_option)
+						
+			else:
+				print (text_format.CRED + "\nThe file {}.txt is empty!\n".format(filename) 
+					+text_format.CEND )
+		else:
+			print (text_format.CRED + "\nThe file {}.txt does not exist!\n".format(filename) 
+				+text_format.CEND)					
+
+		return
```

Comment:
A deeply nested block of code is hard to debug. Avoid it by using not when you can. 

`if not  os.path.isfile(filename + ".txt")`
`return no file duuhhh`

`if not os.stat(filename + ".txt").st_size:`
`return empty file! puuhleeez!`